### PR TITLE
fix!: ANDROID_HOME is the new default, to check first and give advice

### PR DESCRIPTION
**Pull request by ath0mas**
### Platforms affected
Android

### Motivation and Context

cordova-android 11.0.0 (since #1444) reports about `Android SDK: undefined` while `ANDROID_HOME` is set and `ANDROID_SDK_ROOT` is not.

### Description

```
cordova create hello com.example.hello HelloWorld
cd hello/
cordova platform add android@11.0.0
cordova build
```
with environment variables set for Java (`JAVA_HOME`) and Android (`ANDROID_HOME`, and no more `ANDROID_SDK_ROOT`)

results in a successful build but with wrong reporting during the first steps:
```
Checking Java JDK and Android SDK versions
ANDROID_HOME=C:\my\path\to\android-sdk (recommended setting)
ANDROID_SDK_ROOT=undefined (DEPRECATED)
Using Android SDK: undefined
```

### Testing

- creating and building a blank project using local clone of my fork of cordova-android
- `npm run test`

### Checklist

- [x] I've run the tests to see all new and existing tests pass
- [x] I added automated test coverage as appropriate for this change
- [x] Commit is prefixed with `(platform)` if this change only applies to one platform (e.g. `(android)`)
- [x] If this Pull Request resolves an issue, I linked to the issue in the text above (and used the correct [keyword to close issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/))
- [x] I've updated the documentation if necessary

### CODE REVIEW

**Review by breautek — APPROVED** [review]
lgtm -- but I'm not sure if this can be considered a patch.

It could be considered a breaking change... particularly if they did had `ANDROID_HOME` set, but to an invalid value. Therefore, we may have to wait until we prepare a new major release before this can be merged.

**Review by erisu — APPROVED** [review]

**Comment by breautek** [comment]
Final call for review before merging.

**Comment by breautek** [comment]
Thank you @ath0mas for helping us keep up with Android tooling standards.

**Pull request merged**
